lib/web_ui/lib/src/engine/text_editing.dart

```diff
@@ -715,10 +768,16 @@ class HybridTextEditing {
   ///
   /// They are changed depending on the messages coming from method calls:
   /// "TextInput.setStyle", "TextInput.setEditableSizeAndTransform".
+  ///
+  /// In IOS, initial positionig of input element is not done in this method.
+  /// This method changes the location of the input element if it is already
+  /// positioned.
+  /// See [TextEditingElement._delayBeforePositining].
   void _setDynamicStyleAttributes(html.HtmlElement domElement) {
     if (_editingLocationAndSize != null &&
-        !(browserEngine == BrowserEngine.webkit &&
-            operatingSystem == OperatingSystem.iOs)) {
+        (inputPositioned ||
+            !(browserEngine == BrowserEngine.webkit &&
+                operatingSystem == OperatingSystem.iOs))) {
```

Comment:
This condition is becoming harder to understand. I suggest adding getters to make it more readable:
```dart
bool get _canPositionInput => inputPositioned || _doesKeyboardShiftInput;
bool get _doesKeyboardShiftInput =>
    browserEngine == BrowserEngine.webkit && operatingSystem == OperatingSystem.iOs;

void _setDynamicStyleAttributes(html.HtmlElement domElement) {
  if (_editingLocationAndSize != null && _canPositionInput) {
    setStyle(domElement);
  }
}
```

Reply:
Thank you, it looks a lot better now. 

(note: I defined as _canPositionInput => inputPositioned || !_doesKeyboardShiftInput)